app/src/main/java/derpibooru/derpy/server/providers/CommentProvider.java

```diff
@@ -0,0 +1,88 @@
+package derpibooru.derpy.server.providers;
+
+import android.content.Context;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import derpibooru.derpy.data.server.DerpibooruComment;
+import derpibooru.derpy.data.server.DerpibooruFilter;
+import derpibooru.derpy.data.server.DerpibooruTagDetailed;
+import derpibooru.derpy.server.QueryHandler;
+import derpibooru.derpy.server.parsers.CommentParser;
+
+public class CommentProvider extends Provider<DerpibooruComment> {
+    private final DerpibooruFilter mFilter;
+
+    private int mCommentId;
+
+    public CommentProvider(Context context, QueryHandler<DerpibooruComment> handler,
+                           DerpibooruFilter userFilter) {
+        super(context, handler);
+        mFilter = userFilter;
+    }
+
+    /**
+     * Returns a CommentProvider for the particular <strong>comment</strong> ID.
+     *
+     * @param id comment ID
+     */
+    public CommentProvider id(int id) {
+        mCommentId = id;
+        return this;
+    }
+
+    @Override
+    protected Map<String, String> getHeaders() {
+        /* ! important: without the header, the server returns 422. could be a security feature or something, hence
+         * TODO: use a public API method instead of this obscure URI */
+        Map<String, String> headers = new HashMap<>(1);
+        headers.put("x-requested-with", "XMLHttpRequest");
```

Comment:
Rails assumes that the JS format will always be X-Requested-With XMLHttpRequest.
```ruby
  def show
    authorize! :read, Comment
    respond_to do |format|
      format.json { render json: @comment }
      format.all  { render layout: false }
    end
  end
```
Here you can just use the HTML format by requesting the path with no extension, or with the `.html` extension. X-Requested-With is not required.